### PR TITLE
mc: Vendorize minio-go this brings in streaming support.

### PR DESCRIPTION
**Pull request by harshavardhana**
This PR brings following changes

- Brings streaming support for mc uploads.
- Brings support for copying objects upto 5TiB
  as a consequence of ComposeObject implementation
  in minio-go.

### CODE REVIEW

File: cmd/common-methods.go [hunk 1]
```diff
@@ -136,8 +136,8 @@ func uploadSourceToTargetURL(urls URLs, progress io.Reader) URLs {
 	targetURL := urls.TargetContent.URL
 	length := urls.SourceContent.Size
 
-	// Optimize for server side copy if object is <= 5GiB and the host is same.
-	if length <= globalMaximumPutSize && sourceAlias == targetAlias {
```
Comment (vadmeste) [inline cmd/common-methods.go]: Removing `length <= globalMaximumPutSize` because minio-go already does that ?

Comment (harshavardhana) [inline cmd/common-methods.go]: Now minio-go supports copyObject upto 5TiB // cc @donatello

**Comment by donatello** [comment]
PR https://github.com/minio/minio-go/pull/741 also needs to be in minio-go before it is vendored here to be sure that there are no known bugs.

**Comment by harshavardhana** [comment]
> PR minio/minio-go#741 also needs to be in minio-go before it is vendored here to be sure that there are no known bugs.

Yes waiting on two PRs to be merged

**Comment by harshavardhana** [comment]
PR is updated @donatello @krishnasrinivas with the latest minio-go.

**Pull request merged**